data/models.py

```diff
@@ -285,6 +285,10 @@ class Student(models.Model):
     course = models.ForeignKey(Course, on_delete=models.CASCADE, related_name="students")
     key = URLKeyField(max_length=64, help_text="Alphanumeric student id")
 
+    name = models.CharField(max_length=64, blank=True, default='No Name', help_text="Full name of the student")
```

Comment:
Is the default value "No name" really useful? What if the value is just empty instead? Templates could even render the text "No name" if the name variable is empty (if it is really necessary to emphasize that there is no name).

Reply:
Templates create links using the name so fixing the template alternatives would become a requirement. It may or may not also appear in some command line outputs where it could cause confusion. When testing from filesystem all students will become "No Name" as there is no API calls to fetch meta data for submissions.

Reply:
Ok, the default can stay in the model field.